### PR TITLE
LC React refactor: create reusable input components

### PR DESCRIPTION
**Pull request by virginiacc**
### Additions
- New reusable text, select, and radio input components
- Styled select wrapper component

### Removals
- Loan-specific input components

### Changes
- Don't store downpayment-percent on loan. Calculate it for display in downpayment component, and when it changes calculate downpayment value for update on loan.
- When new rates are fetched, call loan store update method so calculated values are generated
- Set generated values directly on loan so they're available for later calculations
- Update loan input table row component to use new input components

### Review
@cfarm

### CODE REVIEW

**Comment by cfarm** [comment]
Started reviewing this; @virginiacc can you walk us through it during pairing tomorrow? I'll plan to check this out locally in the morning and merge, although you'll need to fix conflicts first.

File: src/static/js/modules/loan-comparison/mortgage-calculations.js [hunk 1]
```diff
@@ -14,11 +14,13 @@ mortgage['discount'] = function (loan) {
 };
 
 mortgage['downpayment-percent'] = function (loan) {
-  return Math.round(+loan['downpayment'] / +loan['price'] * 100) || 0;
+    var dp = +loan['downpayment'];
+    var price = +loan['price'];
+    return (dp && price) ? Math.round(dp / price * 100) : 0;
 };
 
 mortgage['downpayment'] = function (loan) {
```
Comment (cfarm) [inline src/static/js/modules/loan-comparison/mortgage-calculations.js]: When you merge master to this branch, check that @amymok's tests pass here.

**Comment by virginiacc** [comment]
@cfarm I've pulled in all the upstream changes & everything seems ok, so this should be ready to merge.

**Comment by virginiacc** [comment]
Oh, on second thought, maybe I'll pull in @amymok's mortgage insurance PR first so I can handle any conflicts...

**Comment by cfarm** [comment]
Let's merge this first, and give us a chance to review hers separately. We can fix any conflicts together if needed

**Pull request merged**